engine/src/main/scala/cromwell/services/EngineMetadataServiceActor.scala

```diff
@@ -1,14 +1,30 @@
 package cromwell.services
 
-import akka.actor.{Actor, ActorLogging}
+import akka.actor.{ActorRef, Actor, ActorLogging}
 import com.typesafe.config.Config
 import cromwell.engine.db.DataAccess
 import cromwell.services.MetadataServiceActor._
+import wdl4s.values.{WdlArray, WdlValue}
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.language.postfixOps
 import scala.util.{Failure, Success}
 
+object EngineMetadataServiceActor {
+
+  implicit class EnhancedServiceRegistryActorForMetadata(val actor: ActorRef) extends AnyVal {
+    def pushWdlValueMetadata(metadataKey: MetadataKey, output: WdlValue): Unit = {
+      output match {
+        case WdlArray(_, valueSeq) =>
```

Comment:
Considering the old metadata... yes I believe so. So predicting the next ticket for the metadata builder, we might need to make sure maps inside arrays [ [ inside maps]* inside arrays]* also work